psutil/arch/osx/process_info.c

```diff
@@ -222,12 +234,43 @@ psutil_get_cmdline(long pid) {
         }
     }
 
-    free(procargs);
-    return py_retlist;
+    // build an environment variable block
+    env_start = arg_ptr;
+
+    py_retenv = PyBytes_FromStringAndSize(NULL, arg_end - arg_ptr);
+
+    if (!py_retenv)
+        goto error;
+
+    memset(PyBytes_AS_STRING(py_retenv), 0, PyBytes_GET_SIZE(py_retenv));
+
+    while (*arg_ptr != '\0' && arg_ptr < arg_end) {
+        char *s = memchr(arg_ptr, '\0', arg_end - arg_ptr);
+
+        if (!s)
+            break;
+
+        memcpy(PyBytes_AS_STRING(py_retenv) + (arg_ptr - env_start), arg_ptr, s - arg_ptr);
+
+        arg_ptr = s + 1;
+    }
+
+    if (_PyBytes_Resize(&py_retenv, arg_ptr - env_start + 1))
+        goto error;
+
+    py_ret = Py_BuildValue("(N, N)", py_retlist, py_retenv);
+
+    if (!py_ret)
+        goto error;
+
+   free(procargs);
+
+   return py_ret;
```

Comment:
Same as for Windows, I would like this to be defined as a separate `psutil_proc_environ` function instead of merging it with cmdline (the merge / refactoring can be done in a separate PR). Also calculating cmdline + environ together is slower thant cmdline alone or environ alone.